janus_server/src/aggregator.rs

```diff
@@ -1091,11 +1086,68 @@ impl VdafOps {
             )
             .await?;
 
+        // At the moment we handle the AggregateShareReq, there could be some incomplete aggregation
+        // jobs whose results not been accumulated into the batch unit aggregations we just queried
+        // from the datastore, meaning we will aggregate over an incomplete view of data, which:
+        //
+        //  * reduces fidelity of the resulting aggregates,
+        //  * could cause us to fail to meet the minimum batch size for the task,
+        //  * or for particularly pathological timing, could cause us to aggregate a different set
+        //    of reports than the leader did (though the checksum will detect this).
+        //
+        // There's not much the helper can do about this, because an aggregate job might be
+        // unfinished because it's waiting on an aggregate sub-protocol message that is never coming
+        // because the leader has abandoned that job.
+        //
+        // Thus the helper has no choice but to assume that any unfinished aggregation jobs were
+        // intentionally abandoned by the leader and service the aggregate share request with
+        // whatever batch unit aggregations are available now.
+        //
+        // See issue #104 for more discussion.
+
         let mut total_report_count = 0;
         let mut total_checksum = [0u8; 32];
         let mut total_aggregate_share: Option<A::AggregateShare> = None;
 
         for batch_unit_aggregation in &batch_unit_aggregations {
+            // Check how many rows in aggregate_share_jobs have a batch interval that includes this
```

Comment:
I don't think these checks are sufficient to protect the privacy of client reports.

Say we have `match_batch_lifetime` = 2, and three consecutive batch units `A`, `B`, `C`. Each of these batch units does not have enough reports to meet the `min_batch_size`, but any two batch units do.

If a collector requested aggregation of `[A_start, C_end)`, followed by `[B_start, C_end)`, the current checks would allow both collections to occur. But then the collector could determine the aggregation of `[A_start, A_end)` by subtracting the second collection from the first, which violates the `min_batch_size` requirements.

One solution to this would be to require the intervals of different collection requests to either match exactly, or not overlap at all. But that may be too restrictive--perhaps there is a more flexible solution?

Reply:
Fair enough; but from a practical standpoint, IMO this is an unresolved privacy vulnerability, so I think we need to resolve it _somehow_ before we can run "real" data through our system (though we might elect for something that we know we are going to throw away later, a la the recent aggregator<->aggregator auth changes).

That said, there's no reason to hold up this PR for it. Would you mind leaving a TODO or filing an issue for now?

Reply:
Wrote this up in #149